app/models/company.rb

```diff
@@ -100,9 +100,7 @@ def shorten_zip_code
   end
 
   def eligible_customer_requests
-    CustomerRequest.where('expires_date >= ?', Date.today()).where(
-      service_category_id: service_categories
-    ).select {|cr| cr.distance_from([latitude, longitude]) <= service_radius }
+    CustomerRequest.where('expires_date >= ? AND service_category_id in (?)', Date.today, service_categories.map(&:id)).select { |cr| cr.distance_from([latitude, longitude]) <= service_radius }.delete_if { |cr| cr.quotes.any? { |quote| quote.company_id == id || quote.accepted == true } || cr.open_quotes.count >= 3 }
```

Comment:
use CustomerRequest.includes(:quotes).where... so avoid doing n+1 queries when you call cr.quotes.any?